Packages/ClientRuntime/Sources/Networking/Http/CRT/CRTClientEngine.swift

```diff
@@ -37,7 +37,7 @@ public class CRTClientEngine: HttpClientEngine {
             logger.error("Server name was not able to be set in TLS Connection Options. TLS Negotiation will fail.")
             logger.error("Error: \(err.localizedDescription)")
         }
-        var socketOptions = SocketOptions(socketType: .stream)
+        let socketOptions = SocketOptions(socketType: .stream)
```

Comment:
You can still adjust connect timeout on iOS even if this is let?

Reply:
Seems that way :)

Reply:
Let me triple check that we can do that on iOS

Reply:
```
        let socketOptions = SocketOptions(socketType: .stream)
#if os(iOS) || os(watchOS)
        socketOptions.connectTimeoutMs = 30_000
#endif
```

Seems to compile with iOS, so i think we are good.  Here is how i tested it.
1.  Compiled the following for mac:
```
        let socketOptions = SocketOptions(socketType: .stream)
#if os(iOS) || os(watchOS)
alkdjfklasjflksjdf
        socketOptions.connectTimeoutMs = 30_000
#endif
```
Result: build successful

1.  Compiled the following for iOS:
```
        let socketOptions = SocketOptions(socketType: .stream)
#if os(iOS) || os(watchOS)
alkdjfklasjflksjdf
        socketOptions.connectTimeoutMs = 30_000
#endif
```
result: build failure

1.  Compiled the following for iOS (removed junk):
```
        let socketOptions = SocketOptions(socketType: .stream)
#if os(iOS) || os(watchOS)
        socketOptions.connectTimeoutMs = 30_000
#endif
```
result: build succeeded

Reply:
nice, lets merge this then!